### ISSUE TITLE
Review whether AddressBook records should be wrapped in `newtype`

### ISSUE THREAD

**Issue by milesfrain** [purescript-contrib/purescript-book#118]
I'm questioning the benefits of wrapping record types in `newtype`. It doesn't seem to be required for anything except custom `Show` instances, which seem to just mimic the default record show.

This is a major annoyance in Chapter 8 (example still in progress to be merged back in), but affects other chapters as well. So a few chapters will need to be modified if this simplification of removing `newtype` is applied.

Here's a summary of the dilemma. Note that this situation is partially improved by using separate state for each field, but it would be even simpler with a single person state and no `newtype` wrapping, and we should strive for simplicity for our beginner readers.

I find myself doing this:
```hs
pp@(Person p@{ homeAddress: Address a }) /\ setPerson <- useState examplePerson

setPerson \_ -> Person p { homeAddress = Address a { street = s } }
```
When I'd rather do this:
```hs
person /\ setPerson <- useState examplePerson

setPerson \_ -> person { homeAddress { street = s } }
```
Here are the wrapped records for reference:
```hs
newtype Person = Person
  { homeAddress :: Address
  }

newtype Address = Address
  { street :: String
  }
```
I'd rather just use basic records:
```hs
type Person =
  { homeAddress :: Address
  }

type Address =
  { street :: String
  }
```

**Comment by JordanMartinez** [comment]
For what it's worth, this seems sensible to me. Newtypes are verbose and if
it slows down the learner when teaching more fundamental things, then
covering that should be postponed until later. Perhaps you could even show
the tradeoffs that come with it?
If removed, when would newtypes be covered in the book?

**Comment by hdgarrood** [comment]
I suspect part of the reason this was done originally was that records previously weren’t able to have instances at all - it was only once RowToList came about that we were able to provide instances like Show or Eq for records.

**Comment by milesfrain** [comment]
> If removed, when would newtypes be covered in the book?

[newtypes are covered in chapter 5](https://book.purescript.org/chapter5.html#newtypes), and then in more detail in ch6 "type classes".

I'll remove the newtype wrappers from the address book example in chapter 8 as a separate PR once the bulk of the changes to restore that section are merged.

**Comment by milesfrain** [comment]
Closing, now that wrapper-free address book code is part of Ch7, Ch8, and soon-to-be Ch10
